### DISCUSSION TITLE
Typed test client

### DISCUSSION THREAD

**Discussion by craigmulligan** [fastapi/fastapi#12491] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.testclient import TestClient
from generated import GeneratedClient
from .main import app

test_client = TestClient(app)
typed_test_client = GeneratedClient(test_client)


def test_read_main():
    response = typed_test_client.get_home()
    assert response.message == "Hello world"
```

### Description

I've looked at [openapi-generator](https://openapi-generator.tech/docs/usage/#generate) it can generate a python client, but it would require some work to make fastapi's `TestClient` satisfy the interface that these generated clients accept. 

Eg `UserApi(generated_http_client)` vs `UserApi(fast_api_test_client)`. 

Just wondering if anyone has tackled this before?

### Operating System

macOS

### FastAPI Version

doesn't matter

### Pydantic Version

doesn't matter

### Python Version

doesn't matter

**Comment by YuriiMotov** [comment]
Seems that this idea is not in high demand.
Let's close this for now to keep the discussions clean.
Feel free to re-open it and provide more reasoning on why we may need this if you still want to discuss it.
